app/models/job_execution.rb

```diff
@@ -173,7 +173,7 @@ def setup!(dir)
       return false unless @repository.setup!(dir, @reference)
       commit = @repository.commit_from_ref(@reference, length: nil)
       tag = @repository.tag_from_ref(@reference)
-      @job.update_git_references!(commit: commit, tag: tag)
+      @job.update_git_references!(commit: commit, tag: tag) if commit || tag
```

Comment:
I added this before I made the change to `commit_from_ref` in an attempt to avoid overwriting `commit` to `nil`.

It might still help in case the user passes in an invalid git ref, causing `@repository.commit_from_ref` above to return `nil`.